qcsubmit/workflow_components/filters.py

```diff
@@ -385,9 +387,11 @@ def _apply(self, molecules: List[Molecule]) -> ComponentResult:
 
         # run the the molecules and calculate the number of rotatable bonds
         for molecule in molecules:
-            if len(molecule.find_rotatable_bonds()) > self.maximum_rotors:
+            if (
```

Comment:
We need some logic to handle the case where `maximum_rotors` or `minimum_rotors` is `None`. Lets also only call `find_rotatable_bonds` once and save the result for the comparison.

Reply:
> @pavankum thanks for making the changes only the last two things to change and this should be good to go.

@JoshHorton My apologies for the delay in fixing these issues. Please review the changes when you have time. Thank you.